### PR TITLE
Adding docker container os and version to default metrics

### PR DESCRIPTION
**Pull request by tcallsen**
## Description

Includes the Docker OS and version metrics read from the [/etc/os-release](https://www.linux.org/docs/man5/os-release.html) file, which is available in both Ubuntu and CentOS docker containers.

Sample `/etc/os-release` file:

```
PRETTY_NAME="Ubuntu 22.04 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

The above file is parsed using [dotenv](https://github.com/motdotla/dotenv#parsing). `NAME` and `VERSION_ID` are used as metrics. 

Sample Metrics produced by IT runs: 
- [New Relic query](https://onenr.io/0Zw0Zr5PNjv)
- [CSV export](https://github.com/adobe/node-openwhisk-newrelic/files/9223488/ASSETS-13042.docker.container.OS.and.version.metrics.csv) - partial export, happy to re-run ITs and get full export if needed

Considerations
- Node's `os.platform()` did not return useful information inside of a Docker container - e.g. response: `#1 SMP PREEMPT Thu Mar 17 17:05:54 UTC 2022`
- Did not think this belonged in the [cgroup metrics library](https://github.com/adobe/node-cgroup-metrics) since that focused on cpu usage, memory, running processes

Fixes # ASSETS-13042

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/metrics.js [hunk 1]
```diff
@@ -99,6 +100,20 @@ function openwhisk(metrics) {
     } catch (e) {
         // ignore error if not in the context of a docker container
     }
+
+    try {
+        // container os and version read from os-release - https://www.linux.org/docs/man5/os-release.html
+        const osReleaseText = fs.existsSync('/etc/os-release') ?
```
Comment (tmathern) [inline lib/metrics.js:106]: Filenames can be turned into constants

**Review by tmathern — APPROVED** [review]
I would just set the filenames as constants. Once this is done, lgtm.

File: test/metrics.test.js [hunk 2]
```diff
@@ -69,6 +69,47 @@ describe("metrics.js", () => {
...
+            assert.equal(metrics.test, 1);
+        });
+
+        it("should ignore container os and version in metrics if unavailable", () => {
+            const metrics = Metrics.openwhisk({ test: 1 });
+            assert.ok(typeof metrics.containerOS === "undefined");
+            assert.ok(typeof metrics.containerOSVersion === "undefined");
+            assert.equal(metrics.test, 1);
+        });
+
```
Comment (pheenomenon) [inline test/metrics.test.js:112]: Would be nice to add a test to ignore any possible errors.

File: lib/metrics.js [hunk 3]
```diff
@@ -99,6 +100,20 @@ function openwhisk(metrics) {
...
+            fs.readFileSync('/etc/os-release') :
+            fs.readFileSync('/usr/lib/os-release');
+        const osRelease = dotenv.parse(osReleaseText);
+        metrics.containerOS = osRelease.NAME || undefined;
+        metrics.containerOSVersion = osRelease.VERSION_ID || undefined;
```
Comment (pheenomenon) [inline lib/metrics.js:111]: is this to help with scenarios where we would have workers with both CentOS and Ubuntu running ? JIRA doesn't say anything.

**Review by pheenomenon — APPROVED** [review]

File: lib/metrics.js (hunk 3 above)
Comment (tmathern) [reply lib/metrics.js:111]: No, it's more to track which one we switched

File: lib/metrics.js (hunk 1 above)
Comment (tcallsen) [reply lib/metrics.js]: Good idea - moved these paths to constants.

Taylor

File: test/metrics.test.js (hunk 2 above)
Comment (tcallsen) [reply test/metrics.test.js:112]: Good idea - I added some checks to ensure the files exist prior to reading them, and a unit test that mocks a file read error to confirm errors are handled appropriately.

Taylor

**Pull request merged**